.changeset/gorgeous-otters-jam.md

```diff
@@ -0,0 +1,8 @@
+---
+'@backstage/plugin-catalog-backend-module-github': patch
+---
+
+Adjust the backend module name/id to match the provider name.
+
+- Renames `githubEntityProviderCatalogModule` to `gitHubEntityProviderCatalogModule`.
+- Renames `moduleId` from `githubEntityProvider` to `githubEntityProvider`.
```

Comment:
These should actually be reverted. We haven't been clear on this but we try to make text say "GitHub" but code say `github` or `Github`. Actually the `GitHubEntityProvider` is named wrong. If you want to do the reverse instead, and deprecate `GitHubEntityProvider` while instead exporting a new `GithubEntityProvider`, then that would be ok too

Reply:
good to know. I will adjust/invert it.

Reply:
I decided to drop that commit and rather do it as separate PR after merging this one to prevent scope creep